### PR TITLE
add LAMP developer chart (WIP)

### PR DESCRIPTION
**Pull request by prydonius**

### CODE REVIEW

File: lamp/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{{- else -}}
+["php", "-S", "0.0.0.0:80"]
+{{- end -}}
+{{ end}}
+
+{{define "lamp.envs"}}
+{{- $type := default "php" .Values.phpFramework -}}
+{{- if eq "laravel" $type -}}
+        - name: APP_KEY
+          value: base64:3XSuN7vqnaL+bL44wng95RXHzCWuXOesIxpjhORYl8A=
```
Comment (prydonius) [inline lamp/templates/_helpers.tpl]: This should be generated

File: lamp/templates/lamp-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        app: {{template "lamp.fullname" .}}
+        heritage: {{ .Release.Service | quote }}
+        release: {{ .Release.Name | quote }}
+        chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    spec:
+      containers:
+      - name: {{template "lamp.fullname" .}}
+        # We use the Bitnami PHP-FPM Container to run the PHP application
+        # https://hub.docker.com/r/bitnami/php-fpm
+        image: prydonius/php-fpm:7.0.6-r1
```
Comment (prydonius) [inline lamp/templates/lamp-deployment.yaml]: Switch back to bitnami/php-fpm once https://github.com/bitnami/bitnami-docker-php-fpm/pull/42 is merged

Comment (sameersbn) [inline lamp/templates/lamp-deployment.yaml]: bitnami/bitnami-docker-php-fpm#42 is merged and part of `bitnami/php-fpm:7.0.6-r1`

/cc @prydonius

File: apache/templates/apache-rc.yaml [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
     provider: apache
+    release: {{ .Release.Name }}
     heritage: bitnami
 spec:
   replicas: 1
   selector:
     provider: apache
-    version: 2.4.20-r0
+    release: {{ .Release.Name }}
+    version: {{ .Values.imageTag }}
```
Comment (technosophos) [inline apache/templates/apache-rc.yaml]: That's a good idea

File: drupal/charts/mariadb/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+Based on the [Bitnami MariaDB](https://github.com/bitnami/bitnami-docker-mariadb) image for docker, this Chart bootstraps a [MariaDB](https://mariadb.com/) deployment on a [Kubernetes](http://kubernetes.io) cluster using [Helm](https://helm.sh).
+
+## Persistence
+
+For persistence of the MariaDB data, mount a [storage volume](http://kubernetes.io/docs/user-guide/volumes/) at the `/bitnami/mariadb` path of the MariaDB pod.
+
+By default the MariaDB Chart mounts an [emptyDir](http://kubernetes.io/docs/user-guide/volumes/#emptydir) volume.
+
+## Configuration
```
Comment (technosophos) [inline drupal/charts/mariadb/README.md]: This section probably needs some changes

File: drupal/charts/mariadb/templates/mariadb-rc.yaml [hunk 5]
```diff
@@ -1,41 +1,43 @@
-#helm:generate helmc tpl -d tpl/values.toml -o manifests/mariadb-rc.yaml $HELM_GENERATE_FILE
 apiVersion: v1
 kind: ReplicationController
 metadata:
-  name: mariadb
+  name: {{ printf "%s-%s" .Release.Name "mariadb" | trunc 24 }}
```
Comment (technosophos) [inline drupal/charts/mariadb/templates/mariadb-rc.yaml]: Out of curiosity, can you replace `"mariadb"` with `.Chart.Name` here?

Comment (prydonius) [inline drupal/charts/mariadb/templates/mariadb-rc.yaml]: Oh! That's a good idea, thanks!

**Comment by prydonius** [comment]
I think this PR got weirdly mixed up with some other changes?

**Comment by sameersbn** [comment]
> I think this PR got weirdly mixed up with some other changes?

Hey @prydonius I had rebased the helm-2.0 branch and it must have messed up your PR. To fix the PR please branch off from helm-2.0 and apply your changes again.

File: drupal/charts/mariadb/README.md (hunk 4 above)
Comment (sameersbn) [inline drupal/charts/mariadb/README.md]: As of now, the README's haven't yet been updated.

**Comment by prydonius** [comment]
Ah - thanks @sameer, I got a hold of your rebased branch and cherry-picked my commit over.

File: lamp/charts/mariadb/templates/_helpers.tpl [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{define "mariadb.name"}}{{default "mariadb" .Values.nameOverride | trunc 24 }}{{end}}
```
Comment (sameersbn) [inline lamp/charts/mariadb/templates/_helpers.tpl]: is this used?

Comment (prydonius) [inline lamp/charts/mariadb/templates/_helpers.tpl]: No, doesn't look like it! I guess the name override is for upgrade purposes or something?

Comment (sameersbn) [inline lamp/charts/mariadb/templates/_helpers.tpl]: I have no clue.

**Pull request closed without merging**